backend-rust/migrations/0001_initialize.up.sql

```diff
@@ -247,6 +247,63 @@ CREATE TABLE bakers(
         BIGINT
 );
 
+-- Every WASM module on chain.
+CREATE TABLE smart_contract_modules(
+    index
+        BIGINT
+        PRIMARY KEY,
+    -- Module reference of the wasm module.
+    module_reference
+        CHAR(64)
+        UNIQUE
+        NOT NULL,
+    -- The absolute block height when the module was deployed.
+    deployment_block_height
+        BIGINT
+        NOT NULL,
+    -- Transaction index in the block deploying the module.
+    deployment_transaction_index
+        BIGINT
+        NOT NULL,
+    -- Embedded schema in the wasm module if present.
+    schema TEXT
```

Comment:
```suggestion
    schema TEXT NOT NULL DEFAULT ''
```
Personally I prefer to keep text columns as not null and use an empty string to indicate a missing field - this means that there is not two "empty" states, i.e. both null and the empty string. This only makes sense when the empty string is not a meaningful value though, which I'm guessing it isn't here but not sure.

Reply:
`NULL` means that there is no embedded schema in the module.

It is possible to compile the wasm module (with embedded schema), while specifying no `schema` in the smart contract. In case that is represented as empty string, I keep the current behavior.